### PR TITLE
feat: support building centos/debian Docker images in multiplatform format

### LINKED ISSUE #1089: Maintain our own images for centos and debian
Because of ARM support, we cannot simply pull the images from Docker hub on ARM machines, as those images where built only for AMD.

We'd need to maintain our own Dockerfiles and build them in both AMD and ARM.

cc/ @elastic/observablt-robots

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR adds a pipeline to build and push the Docker images for Centos and Debian (used by the elastic-agent) in multiplatform format: ARM64 and AMD64.

To achieve that we:

- copied the Dockerfiles from the upstream project to this repository
- created a script for building and pushing the images for each OS (centos and debian). Each image's Docker name will contain the platform as suffix: `centos-systemd-arm64:latest`
- created a script for pushing the multiplatform manifest for each image (centos and debian), using Infra's `manifest-tool`. This tool is bundled into a Docker image.

>This script leverages infra's tool to write the manifest, which needs to be ran right after the images have been built and pushed. Therefore, the tool will write the manifest for both platforms (AMD/ARM), inspecting the existing platform-specific repositories, combining them into the target.

- create a regular pipeline to orchestrate the build&push. This pipeline has 4 different stages:
  1. git checkout the e2e-testing repo
  2. build & push the AMD images, grabbing an AMD worker (ubuntu-20)
  3. build & push the ARM images, grabbing an ARM worker
  4. push the multiplatform manifest

## Why is it important?
In order to ran the Elastic Agent in ARM, we need to have Centos and Debian containers for ARM. This is not the case of integrations, which can be run elsewhere.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ] Pipeline scheduler
- [ ] JJBB

## How to test this PR locally

It's possible to run the scripts to build&push the images, but a `docker login` must be performed first. The multiplatform manifest needs the ARM images to be created.

## Related issues

- Relates #707
- Closes #1089

### CODE REVIEW

File: .ci/build-docker-images.groovy [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'ubuntu-20' }
+  environment {
+    REPO = 'e2e-testing'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    DOCKER_REGISTRY = 'docker.elastic.co'
+    DOCKER_REGISTRY_SECRET = 'secret/observability-team/ci/docker-registry/prod'
+    HOME = "${env.WORKSPACE}"
```
Comment (kuisathaverat) [inline .ci/build-docker-images.groovy:12]: if you provision agents this variable could not be valid, you should move it to the context of each agent provision.

File: .ci/build-docker-images.groovy [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}",
+          branch: "${params.BRANCH_REFERENCE}",
+          repo: "https://github.com/elastic/${REPO}.git",
+          credentialsId: "${JOB_GIT_CREDENTIALS}"
+        )
+        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+      }
+    }
+    stage('Build AMD Docker images'){
```
Comment (mdelapenya) [inline .ci/build-docker-images.groovy:42]: The build stages could be run in parallel, maybe in a follow-up

File: .ci/build-docker-images.groovy (hunk 1 above)
Comment (kuisathaverat) [reply .ci/build-docker-images.groovy:12]: forget is defined on each agent

File: cli/config/compose/services/centos-systemd/docker-compose.yml [hunk 3]
```diff
@@ -4,6 +4,7 @@ services:
     image: centos/systemd:${centos_systemdTag:-latest}
```
Comment (mdelapenya) [inline cli/config/compose/services/centos-systemd/docker-compose.yml:4]: I'm not changing the images yet. They will be eventually replaced with:

```
image: docker.elastic.co/observability-ci/centos-systemd:latest
```

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**